### PR TITLE
Clustering in Machine Learning Tutorial (V 2.0)

### PR DESCRIPTION
**Pull request by khanteymoori**
Hi,

This PR is a new version of https://github.com/galaxyproject/training-material/pull/1850
I had problems with my branches in the local repository so I created a fresh branch for it.
In this new PR:
Anup comments are applied.
The problem with one figure is solved. 
The introduction part is extended.
Two important parameters in DBSCAN are described.
Data-library.yml file is modified.

@anuprulez, @bgruening please let me know your comments.

### CODE REVIEW

File: topics/statistics/tutorials/clustering_machinelearning/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+ > >
+ > {: .solution }
+ >
+ {: .question }
+
+Other dissimilarity measures exist such as correlation-based distances, which are widely used for gene expression data analyses. Correlation-based distance considers two objects to be similar if their features are highly correlated, even though the observed values may be far apart in terms of Euclidean distance. The distance between two objects is 0 when they are perfectly correlated. Pearson’s correlation is quite sensitive to outliers. This does not matter
+when clustering samples, because the correlation is over thousands of genes. During clustering genes, it is important to be aware of the possible impact of outliers. This
+can be mitigated by using Spearman’s correlation instead of Pearson’s correlation.
+
+# Hierarchial Clustering
```
Comment (anuprulez) [inline topics/statistics/tutorials/clustering_machinelearning/tutorial.md:111]: Should be `Hierarchical ...`

File: topics/statistics/tutorials/clustering_machinelearning/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,484 @@
...
+{: .hands_on}
+
+If you view the result table, you can see the last column is the label for each cluster and as you see, all the setosa samples are grouped in one cluster and two other species (versicolor and virginica) are grouped in the second cluster. From Figure 3 it is obvious that versicolor and virginica are more similar to each other.
+
+## Visulaze the clustering results
+
+The resulting candidate clusterings can be visualized using the Scatterplot w ggplot2 tool. Each sample is color-coded based on its clustering for that sample. 
+Let's visualize the clustering results to see how groups have been built.
+
+> ### {% icon hands_on %} Hands-on: Visualize hierarchical clutsreing result
```
Comment (anuprulez) [inline topics/statistics/tutorials/clustering_machinelearning/tutorial.md:231]: Should be `clustering`

File: topics/statistics/tutorials/clustering_machinelearning/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,484 @@
...
+
+For example, in biology, clustering is often one of the first steps in gene expression analysis and is used to find groups of genes with similar expression patterns.
+
+We represent an observation as an n-dimensional vector. For example, assume that your training data consists of the samples plotted in the following Figure 1:
+
+>    ![data](images/data_before_clustering.png "Sample data before clustering")
+
+Clustering reveal the following four groups, indicated by different colors:
+
+>    ![data](images/data_after_clustering.png "Sample data after clustering")
```
[the remark this answers is not in the document]
Comment (khanteymoori) [reply topics/statistics/tutorials/clustering_machinelearning/tutorial.md:39]: I created new images and replace them

**Comment by khanteymoori** [comment]
Thanks to @anuprulez , the tutorial has been modified using your comments.

File: topics/statistics/tutorials/clustering_machinelearning/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,484 @@
...
+{: .hands_on}
+
+If you view the result table, you can see the last column is the label for each cluster and as you see, all the setosa samples are grouped in one cluster and two other species (versicolor and virginica) are grouped in the second cluster. From Figure 3 it is obvious that versicolor and virginica are more similar to each other.
+
+## Visulaze the clustering results
+
+The resulting candidate clustering can be visualized using the Scatterplot w ggplot2 tool. Each sample is color-coded based on its clustering for that sample. 
+Let's visualize the clustering results to see how groups have been built.
+
+> ### {% icon hands_on %} Hands-on: Visualize hierarchical clutsreing result
```
Comment (anuprulez) [inline topics/statistics/tutorials/clustering_machinelearning/tutorial.md:231]: typo I guess

File: topics/statistics/tutorials/clustering_machinelearning/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,484 @@
...
+>        - {% icon param-select %} *"Choose how to select data by column"*: `All columns EXCLUDING some by column header name(s)`
+>                - {% icon param-text %} *"Type header name(s)"*: `Species`
+>        - {% icon param-select %} *"Clustering Algorithm"*: `KMeans`
+>        - In *"Advanced option"*    
+>                - {% icon param-text %} *"Number of clusters"*: `2`
+> 2. Rename the generated file to `k-means clustering`
+{: .hands_on}
+
+
+> ### {% icon hands_on %} Hands-on: Visualize k-means clutsreing result
```
Comment (anuprulez) [inline topics/statistics/tutorials/clustering_machinelearning/tutorial.md:289]: typo

File: topics/statistics/tutorials/clustering_machinelearning/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,484 @@
...
+
+>    ![data](images/circles_clustering.png "Plot of Clustering Algorithms on Circles Data")
+
+>    ![data](images/moon_clustering.png "Plot of Clustering Algorithms on Moon Data")
+
+
+# Conclusion
+
+In this tutorial, we discussed the clustering algorithms which is used to discover structure in unlabeled data. You learned about the hierarcial, k-means and DBSCAN algorithms. By following these steps, we learned how to perform clustering and visualize results using clustering and plotting tools respectively in Galaxy. There are many other clustering approaches which can be tried out on these datasets to find how they perform. Different datasets can also be analysed using these algorithms. The clustering algorithms have some parameters which can be altered while performing the analyses to see if they affect the clustering or not. In using clustering algorithms, we need to take care of some important aspects like treating outliers in our data and making sure each cluster has sufficient population. Some data pre-processors can also be used to clean the datasets.
```
Comment (anuprulez) [inline topics/statistics/tutorials/clustering_machinelearning/tutorial.md:475]: In general, please do a spelling check. Thanks!

Comment (khanteymoori) [reply topics/statistics/tutorials/clustering_machinelearning/tutorial.md]: Thanks Anup, typos have been corrected.

**Comment by anuprulez** [comment]
:+1:

**Pull request merged**